propka/molecular_container.py

```diff
@@ -48,10 +48,10 @@ def __init__(self, parameters, options=None):
 
     def top_up_conformations(self):
         """Makes sure that all atoms are present in all conformations."""
-        for name in self.conformation_names:
-            if (name != '1A' and (len(self.conformations[name])
-                                  < len(self.conformations['1A']))):
-                self.conformations[name].top_up(self.conformations['1A'])
+        for name in self.conformation_names[1:]:
+            first = self.conformations[self.conformation_names[0]]
```

Comment:
Shouldn't this line be elevated outside the loop rather than repeated N times?
Or does top_up modify self.conformation_names[0]?

Reply:
This was my lazy (I mean: concise :-)) way to avoid checking for `len(self.conformation_names) > 0`. I think the performance cost of N times repetition is not significant.

Reply:
We could make `len(self.conformation_names) > 0` a `@pre` condition. The only function call to this method (in `input.py`) is indeed not reachable if there are no conformations.

Reply:
You don't have to use `len(self.conformation_name) > 0` anymore. The pythonic way is just `if self.conformation_name:`

https://www.geeksforgeeks.org/python-check-if-list-empty-not/